### PR TITLE
Add remaining missing docs to System.Runtime.CompilerServices

### PR DESCRIPTION
**Pull request by krwq**
Fixes: https://github.com/dotnet/runtime/issues/54879

Adding remaining missing docs for System.Runtime.CompilerServices.Unsafe from manual xml file (for ilproj) from dotnet/runtime.

### CODE REVIEW

File: xml/System.Runtime.CompilerServices/Unsafe.xml [hunk 1]
```diff
@@ -920,11 +920,11 @@
...
       </Parameters>
       <Docs>
-        <typeparam name="T">To be added.</typeparam>
-        <param name="source">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <typeparam name="T">The type of the reference.</typeparam>
+        <param name="source">The reference to check.</param>
+        <summary>Returns if a given reference to a value of type <typeparamref name="T"/> is a null reference.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/Unsafe.xml:925]:
```suggestion
        <summary>Determines if a given reference to a value of type <typeparamref name="T"/> is a null reference.</summary>
```

File: xml/System.Runtime.CompilerServices/Unsafe.xml [hunk 2]
```diff
@@ -920,11 +920,11 @@
...
+        <returns><see langword="true" /> if <paramref name="source" /> is a null reference; otherwise, <see langword="false" />.</returns>
+        <remarks>This check is conceptually similar to "(void*)(&amp;source) == nullptr".</remarks>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/Unsafe.xml:927]:
```suggestion
        <remarks>This check is conceptually similar to <code>(void*)(&amp;source) == nullptr</code>.</remarks>
```

File: xml/System.Runtime.CompilerServices/Unsafe.xml [hunk 3]
```diff
@@ -948,9 +948,9 @@
       </TypeParameters>
       <Parameters />
       <Docs>
-        <typeparam name="T">To be added.</typeparam>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <typeparam name="T">The type of the reference.</typeparam>
+        <summary>Returns a reference to a value of type <typeparamref name="T"/> that is a null reference.</summary>
+        <returns>Reference to a value of type <typeparamref name="T"/> that is a null reference.</returns>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/Unsafe.xml:953]:
```suggestion
        <returns>A reference to a value of type <typeparamref name="T"/> that is a null reference.</returns>
```

**Review by gewarren — APPROVED** [review]
Not sure if `&amp;` should just be `&` if it's code-fenced.

**Comment by krwq** [comment]
`&amp;` must work because it's an XML syntax. `&` might not work depending on the tool and parser setting. I do not believe `<code>` will make any difference. I think it's safer to stick with `&amp;`

**Pull request merged**